### PR TITLE
Add devise starter files

### PR DESCRIPTION
**Pull request by Chimoneg27**
For this feature me and my partner did the following

- Made sure you can register a new user.
- User can login with a combination of email and password.
- Made sure hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<h2>Log in</h2>
```
Comment (Lucas-Masaba) [inline app/views/devise/sessions/new.html.erb]: - [ ] According to the project requirements, the log in and sign up page should follow the given wireframe.
### Required
image
### Your Implementation
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Chimoneg27  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
-  Linter checks passing :heavy_check_mark: 
- Successfully added devise :gear: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231214151708_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
```
Comment (V-Blaze) [inline db/migrate/20231214151708_add_devise_to_users.rb:16]: -  You have done great so far :+1: :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: 
- [ ] When I run `rails db:migrate`, I Got these errors shown in the screenshot below. The error points out the email does not exist. Looking at your `AddDeviseToUsers` migration file, it seems you are missing the `email` column which is a required column. I suggest you update this migration file to include the email column. You can add the email column code above `line 7`. 
```ruby
t.string :email,              null: false, default: ""
```

- - Error screenshot:
Selection_068

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.text_field :name, autofocus: true, autocomplete: "name", class: "input" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :bio %><br />
```
Comment (V-Blaze) [inline app/views/devise/registrations/new.html.erb]: Selection_069

- [ ] As stated in the project requirement and pointed out by the previous reviewer, you need to style your devise screens `Login and Registration` page to look like the wireframe. :+1: You have styled the login screen but the `signup page` has not been styled. I'll suggest you style the signup page to look like the wireframe. :wave: :small_airplane: 

- - Your current Design:
Screenshot from 2023-12-15 01-32-44

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Chimoneg27, @erickma1 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Added devise to gem file✔
-  Correct git flow👋 
-  Meaningful commit messages✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231214151708_add_devise_to_users.rb (hunk 2 above)
Comment (Chimoneg27) [reply db/migrate/20231214151708_add_devise_to_users.rb:16]: You don't need to run these. It works fine on my side. That process took me hours to fix.

**Comment by bellom** [comment]
Hello @Chimoneg27 @erickma1 👋,

_This is an additional review as a result of the second opinion request._

I confirm I got the same error as mentioned by @V-Blaze, we can't run your project following your project setup on your README file, however, I can run the project with the following command `rails db:reset`.

## Request Changes ♻️ 

**About the design, you have done well with the login page, however, you still need to work on the sign-up page.**

Kindly work on the sign-up page, I'm sure the next CR will approve.

Keep rocking 🚀!

**Comment by Chimoneg27** [comment]
@bellom I've pushed

**Comment by bellom** [comment]
@Chimoneg27 - kindly give me a few minutes, I will check and revert shortly.

**Comment by bellom** [comment]
Hi @Chimoneg27 @erickma1

## Approved ✅ 

Great work addressing all the requested modifications from the prior code reviews!

Pleas go ahead and merge 🚀 

Keep rocking 🚀!

**Comment by Chimoneg27** [comment]
> Hi @Chimoneg27 @erickma1
> 
> ## Approved ✅
> Great work addressing all the requested modifications from the prior code reviews!
> 
> Pleas go ahead and merge 🚀
> 
> Keep rocking 🚀!

thank you

**Pull request merged**